packages/@react-spectrum/searchfield/src/SearchField.tsx

```diff
@@ -62,6 +62,7 @@ function SearchField(props: SpectrumSearchFieldProps, ref: RefObject<TextFieldRe
   return (
     <TextFieldBase
       {...otherProps}
+      isHovered={isHovered}
```

Comment:
Can TextFieldBase handle isHovered so that we don't need to get it from useSearchField or apply it in SearchField? I imagine this would help cover TextArea/Textfield

Reply:
Yes! I'm gonna fix it to use `useHover` by itself.
However, If I am right, SearchInput needs to pass `isHovered` down to TextFieldBase since it's search-icon needs hover styling as well.

Reply:
I think the search icon is just passed to TextFieldBase, which should handle applying the hovered style to any icon it gets?

Reply:
@snowystinger 
Oh, I see, and it would be perfect! 
But to look at CSS, I think TextField itself doesn't handle its icon's hovered style but SearchField does. 
So I'm thinking about adding some CSS for TextField.

```
/* current */
.spectrum-Search-input {
  /* The value of color is identical for hover/active/focus-ring, but we repeat it here in case one is changed in the future */
  &:hover {
    & ~ .spectrum-Search-icon {
      color: var(--spectrum-search-icon-color-hover);
    }
  }

  &:active {
    & ~ .spectrum-Search-icon {
      color: var(--spectrum-search-icon-color-down);
    }
  }

  &:focus-ring {
    & ~ .spectrum-Search-icon {
      color: var(--spectrum-search-icon-color-key-focus);
    }
  }

  &:disabled {
    & ~ .spectrum-Search-icon {
      color: var(--spectrum-textfield-text-color-disabled);
    }
  }
}
```

```
/* case 1 */

.spectrum-Textfield-input {
  &:hover {
    & ~ .spectrum-Textfield-icon {
      color: var(--spectrum-textfield-placeholder-text-color-hover);
    }
  }
}

/* case 2 */

.spectrum-Textfield-icon {
  color: var(--spectrum-textfield-icon-color);

  &.disabled {
    color: var(--spectrum-textfield-text-color-disabled);
  }

  &:hover {
    color: var(--spectrum-textfield-placeholder-text-color-hover);
  }
}
```

I prefer case 1 to case 2.

Reply:
O thanks for pointing that out. It seems like it's a mistake we have not to put the icon styles completely into the basetextfield given that it'd affect textarea/textfield/searchfield. We can address that later, should allow us to get rid of the spectrum-Search-icon class. Don't worry about it here. Thanks for looking into it!
https://github.com/adobe/react-spectrum/issues/790